src/components/MovieDetailPage.js

```diff
@@ -1,40 +1,80 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Link, withRouter } from 'react-router-dom';
+import API_ROOT_URL from '../constants';
+import NotFoundPage from './NotFoundPage';
+import Spinner from './Spinner';
 
-function MovieDetailPage({ movie, onGoBack }) {
-  const handleGoBack = () => {
-    onGoBack();
+class MovieDetailPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movie: {},
+      error: false,
+      isLoading: true,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchMovie();
+  }
+
+  fetchMovie = async () => {
+    const { match } = this.props;
+    const movieApiUrl = `${API_ROOT_URL}/${match.params.id}`;
+
+    try {
+      const res = await fetch(movieApiUrl);
+      if (!res.ok) {
+        this.setState({ error: true, isLoading: false });
+        throw new Error('Incorrect response');
```

Comment:
Instead of using `boolean` flag, you could set the error message and display it. But that's just an option.

Reply:
I kept it as is since this is now part of a separate apiService

Reply:
I kept it as is since this is now part of a separate apiService